tbxforms/fields.py

```diff
@@ -179,11 +179,12 @@ def compress(self, data_list):
                 fields. If any of the field are blank then None is returned.
 
         """
-        day, month, year = data_list
-        if day and month and year:
-            try:
-                return date(day=int(day), month=int(month), year=int(year))
-            except ValueError as e:
-                raise ValidationError(str(e)) from e
-        else:
-            return None
+        if len(data_list) == 3:
```

Comment:
what happens when one passes just one or two instead of all 3 or no values?

Reply:
Along with the changes that https://github.com/torchbox/tbxforms/pull/29 adds, in both cases:

- if `required=False`: nothing
- if `required=True` and only one of the three fields are completed: separate `ValidationError`s are raised for each individual field that's missing (e.g. "Please enter a month" and  "Please enter a year" if only the day is provided)
- if `required=True`: separate `ValidationError`s are raised for each individual field (e.g. "Please enter a month")

It's probably easier to look at the release branch for this with all of the changes combined: https://github.com/torchbox/tbxforms/pull/52/files#diff-e737fc521b5da51cd98b30ac80c551f4891fc4949dbf297f4da8b26b84490de1